cli/src/ui.rs

```diff
@@ -266,9 +274,59 @@ pub enum PaginationChoice {
     Auto,
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, serde::Deserialize)]
+#[serde(rename_all(deserialize = "kebab-case"))]
+pub enum StreampagerAlternateScreenMode {
+    Always,
+    Never,
+    IfLongOrSlow,
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, serde::Deserialize)]
+#[serde(rename_all(deserialize = "kebab-case"))]
+enum StreampagerWrappingMode {
+    None,
+    Word,
+    Anywhere,
+}
+
+impl From<StreampagerWrappingMode> for streampager::config::WrappingMode {
+    fn from(val: StreampagerWrappingMode) -> Self {
+        use streampager::config::WrappingMode;
+        match val {
+            StreampagerWrappingMode::None => WrappingMode::Unwrapped,
+            StreampagerWrappingMode::Word => WrappingMode::WordBoundary,
+            StreampagerWrappingMode::Anywhere => WrappingMode::GraphemeBoundary,
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, serde::Deserialize)]
+#[serde(rename_all(deserialize = "kebab-case"))]
+struct StreampagerConfig {
+    alternate_screen: StreampagerAlternateScreenMode,
+    long_or_slow_delay_millis: u64,
+    wrapping: StreampagerWrappingMode,
+}
+
+impl StreampagerConfig {
+    fn streampager_interface_mode(&self) -> streampager::config::InterfaceMode {
+        use streampager::config::InterfaceMode;
+        use StreampagerAlternateScreenMode::*;
+        match self.alternate_screen {
+            // InterfaceMode::Direct not implemented
+            Always => InterfaceMode::FullScreen,
+            Never => InterfaceMode::Hybrid,
+            IfLongOrSlow => InterfaceMode::Delayed(std::time::Duration::from_millis(
+                self.long_or_slow_delay_millis,
```

Comment:
nit: no idea if we can generalize this mode. If it's not common, maybe we can omit the `Delayed` mode?

Reply:
I'm undecided. 

One concern I have is that even if we remove it now, if we also simplify the config option names accordingly, it would be difficult to add the Delayed mode back in if we need it. Perhaps this could be solved somehow.

For me, the upshot of the Delayed mode is that it addresses the downside of the default mode when using it to read long texts (e.g. manual pages) where it leaves the last page of what you were reading on on the screen, while still being useable for short outputs. I am not sure how many people would like/want this tradeoff.

A part of this problem could be addressed by #5311, making the Delayed mode less important.